src/backend/crossterm.rs

```diff
@@ -127,6 +119,194 @@ where
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use ratatui::backend::CrosstermBackend;
+    ///
+    /// let terminal = CrosstermBackend::stdout().into_terminal_with_defaults()?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    pub fn into_terminal_with_defaults(self) -> io::Result<Terminal<Self>> {
+        let backend = self.with_raw_mode()?.with_alternate_screen()?;
+        Terminal::new(backend)
+    }
+
+    /// Enables raw mode for the terminal.
+    ///
+    /// Returns an [`io::Result`] containing self so that it can be chained with other methods.
+    ///
+    /// Raw mode is restored when the `CrosstermBackend` is dropped.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use ratatui::backend::CrosstermBackend;
+    ///
+    /// let backend = CrosstermBackend::stdout().with_raw_mode()?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    pub fn with_raw_mode(mut self) -> io::Result<Self> {
+        self.restore_raw_mode_on_drop = true;
+        terminal::enable_raw_mode()?;
+        Ok(self)
+    }
+
+    /// Enables raw mode for the terminal and switches to the alternate screen.
+    ///
+    /// Returns an [`io::Result`] containing self so that it can be chained with other methods.
+    ///
+    /// Alternate screen is restored when the `CrosstermBackend` is dropped.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use ratatui::backend::CrosstermBackend;
+    ///
+    /// let backend = CrosstermBackend::stdout().with_alternate_screen()?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    pub fn with_alternate_screen(mut self) -> io::Result<Self> {
+        self.restore_alternate_screen_on_drop = true;
+        execute!(self.writer, terminal::EnterAlternateScreen)?;
+        Ok(self)
+    }
+
+    /// Enables mouse support for the terminal.
+    ///
+    /// Returns an [`io::Result`] containing self so that it can be chained with other methods.
+    ///
+    /// Mouse support is restored when the `CrosstermBackend` is dropped.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use ratatui::backend::CrosstermBackend;
+    ///
+    /// let backend = CrosstermBackend::stdout().with_mouse_support()?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    pub fn with_mouse_support(mut self) -> io::Result<Self> {
+        self.restore_mouse_capture_on_drop = true;
+        execute!(self.writer, crossterm::event::EnableMouseCapture)?;
+        Ok(self)
+    }
+
+    /// Resets the terminal to its default state.
+    ///
+    /// Disables raw mode, disables mouse capture, and leaves the alternate screen.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use std::io::stdout;
+    ///
+    /// use ratatui::backend::CrosstermBackend;
+    ///
+    /// CrosstermBackend::reset(stdout())?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    pub fn reset(mut writer: W) -> io::Result<()> {
```

Comment:
`CrosstermBackend::stdout()` works without a stdout reference. Reset doesnt.
As the reference is required anyway, I think I would remove the `CrosstermBackend::stdout()` (and stderr) as the import is needed for reset in the end anyway.

Reply:
One possible approach to this might be to add reset to the Backend trait with no parameter and have a version with the parameter on the method which will be called on CrosstermBackend. This overloads the name, but it's reasonable-ish.

Reply:
This is not symmetrical. Personally I find this confusing. Either there is a shortcut / helper for both, or for none. But having only one for creation and not for ending it seems strange.

Reply:
This method is necessary to make it possible to be called from a panic handler and to avoid having to write the same code everywhere. It doesn't need to keep state. It hangs off the Crossterm struct as a convenience to be findable when users are using code completion in their editors.

What about this do you find confusing?

Reply:
Clarification comment added in cd525c335

Reply:
The following uses the same stdout in both places.
```rust
use std::io::stdout;
let backend = CrosstermBackend::new(stdout());
backend.reset(stdout());
```

 For reset the code already needs to import to stdout. There is not much benefit of having `CrosstermBackend::stdout` when the import is already needed in the first place. And it results in a asymetric design: at one place there is a helper for it, at the other there is not. Either both should have a helper for stdout or neither place should have.

Reply:
guess I have misread the method header. reset is `CrosstermBackend::reset(writer)` without a reference to self… so the idea is to only use it for the panic handler and nowhere else…? Ok, so pure drop implementation all the time and nothing else.

Reply:
yes